terragraph/meta-terragraph
# feat(ci): Add sync-source-mirrors job after successful x86 build

## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

Add downloaded tarballs to the shared `source_mirrors` filesystem on
self-hosted runners. Also add BB_GENERATE_MIRROR_TARBALLS = "1" to
bitbake configuration by default for all meta-layers so that tarballs
are generated.

Some parts of the terragraph-image-x86.yml workflow are reformatted, and the file is renamed to remove the word "build"

## Test Plan

1. Delete facebook/openr GitHub tarball from source mirrors
2. Clear openr build artifacts from sstate-cache
3. Run workflow: https://github.com/terragraph/meta-terragraph/actions/runs/2794158940
4. See new openr tarball in source_mirrors: 

```
ubuntu@ip-10-0-20-186:~/source_mirrors$ ls -lh | grep openr -A 2 -B 2
-rw-r--r-- 1 ubuntu ubuntu  6.7M May 10 18:49 git2_github.com.facebook.fbzmq.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu   52M May 10 18:49 git2_github.com.facebook.folly.git.tar.gz
-rw-rw-r-- 1 ubuntu ubuntu   30M Aug  4 05:18 git2_github.com.facebook.openr.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu   21M May 10 18:49 git2_github.com.facebook.proxygen.git.tar.gz
-rw-r--r-- 1 ubuntu ubuntu  6.9M May 10 18:49 git2_github.com.facebook.wangle.git.tar.gz
```
